### ISSUE TITLE
name initials in all caps string gets incorrectly lower cased

### ISSUE THREAD

**Issue by foobargeez** [ap/titlecase#5]
Does *NOT* work:

```$> perl
use Lingua::EN::Titlecase::Simple qw(titlecase);

my $str = 'HAHA AND HEHE A.B TEAM UP';
print titlecase($str), "\n";

Haha and Hehe a.b Team Up
$>```

Thanks!

**Comment by foobargeez** [comment]
@ap would you be interested/willing to look into this? Thanks!

**Comment by ap** [comment]
Yes, I’ve meant to reply for a while. Sorry you’re having to wait.

Uhm, what is your expected output here? “A.b” or “A.B”? Can you provide a non-gibberish example sentence that illustrates the case?

**Comment by foobargeez** [comment]
> Uhm, what is your expected output here? “A.b” or “A.B”?

A.B

> Can you provide a non-gibberish example sentence that illustrates the case?

Oh, no, my example is not gibberish :-)  Anyway, how about this?

Input: `MICHAEL AND JOHN A.P TEAM UP TO DO SOMETHING AWESOME!`
Output (current): `Michael and John a.p Team Up to Do Something Awesome!`
Output (desired): `Michael and John A.P Team up to Do Something Awesome!`

**Comment by ap** [comment]
I guess that helps, insofar as it tells me that a simple example doesn’t make it obvious what’s going on linguistically. This “A.P” with a missing trailing period is not something I have ever seen or would normally write. How does it chunk? Is one person called “Michael” and the other “John A. P”? Are they “Michael A. P.” and “John A. P.”? Is “A.P” some kind of oddball nonstandard personal style (kinda like the people who insist their names be written in lowercase)?

**Comment by foobargeez** [comment]
Thanks!  I don't so much have control on the input text (often received from external sources).  That being said, the way I saw the input above as - one person is called "Michael" and the other "John A.P" but I agree with you that there are other ways to look at it.

Just to make sure this isn't really a corner case, I tried the following online titlecase conversions and they all seem to match the desired output:

- http://titlecase.com/
- http://individed.com/code/to-title-case/
- https://capitalizemytitle.com/

> Is “A.P” some kind of oddball nonstandard personal style (kinda like the people who insist their names be written in lowercase)

Hmm ... I guess I didn't run into such a case with my input names at the least (about 10K names and counting) - who knows what could be next? :-)

**Comment by ap** [comment]
> Just to make sure this isn't really a corner case, I tried the following online titlecase conversions and they all seem to match the desired output:
> 
> * http://titlecase.com/

This one doesn’t implement Gruber’s rules at all.

> * http://individed.com/code/to-title-case/

This one, when given all-caps input, preserves most words in all-uppercase, and when given all-lowercase input, matches the “a.p” output of this code.

> * https://capitalizemytitle.com/

This one, [when switched to NY Times rules](https://capitalizemytitle.com/style/NYTimes/), matches the “a.p” output of this code.

The same is true for [John Gruber’s latest published version (from 2015 as of this writing)](https://gist.github.com/gruber/9f9e8650d68b13ce4d78).

I’m inclined to reject this one…

However: if John himself agrees with the suggestion, then I will accept it with no questions asked. So you can try pinging him about it, or wait until I get around to it. (I want to ask him whether he has made any further changes to his code in the 9 years since, anyway.)
